bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/server/Server.java

```diff
@@ -0,0 +1,274 @@
+            }
+        } catch (Exception e) {
+            logger.warn("Error stopping Communication. {}", e.getMessage());
+        }
+    }
+
+    private void handleCommunication() {
+        try {
+            logger.debug("Waiting for connection in port {}.", this.getPort());
+            IDataPointChangeListener listener = this.changeListener;
+            if (listener != null) {
+                listener.connectionStatusChanged(ThingStatus.OFFLINE);
+            }
+            ServerSocket serverSock = new ServerSocket(this.getPort());
+            this.serverSocket = serverSock;
+            Socket clientSocket = serverSock.accept();
+            this.client = clientSocket;
+            logger.debug("Connection from Partner established {}", clientSocket.getRemoteSocketAddress());
+            if (listener != null) {
+                listener.connectionStatusChanged(ThingStatus.ONLINE);
+            }
+
+            this.startRetries = 0;
+            this.sendUpdateCommand();
+            while (!this.isInterrupted()) {
+                byte[] bytes = getBytesFromInputStream(clientSocket.getInputStream());
+                ArrayList<byte[]> packages = this.getPackages(bytes);
+
+                for (byte[] pack : packages) {
+                    logger.debug("Data received: {}", this.printBytes(pack));
+                    KnxNetFrame frame = KnxNetFrame.createKnxNetPackage(pack, pack.length);
+                    if (frame != null) {
+                        byte[] answer = frame.createFrameAnswer();
+                        if (answer.length > 0) {
+                            this.sendData(answer);
+                        }
+
+                        for (SetDatapointValueMessage message : frame.getValueMessages()) {
+                            logger.debug("Message received: {} {}", message.getId(),
+                                    this.printBytes(message.getData()));
+
+                            IDataPoint dataPoint = null;
+                            for (int k = 0; k < this.dataPoints.size(); k++) {
+                                IDataPoint dp = this.dataPoints.get(k);
+                                if (dp.getId() == message.getId()) {
+                                    dataPoint = dp;
+                                    break;
+                                }
+                            }
+
+                            if (dataPoint != null) {
+                                dataPoint.processData(message.getData());
+                                logger.debug("{} {}", dataPoint.getDescription(), dataPoint.getValueText());
+                                if (listener != null) {
+                                    listener.dataPointChanged(new DataPointChangedEvent(this, dataPoint));
+                                }
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (Exception e) {
```

Comment:
See above

Reply:
I had changed this to IOException as well as most likely any Exception is triggered by the socket communication. Additionally I had to update the method `KnxNetFrame.createKnxNetPackage` to handle the `IllegalArgumentException`. I'll check the binding later tonight and upload the changes in case everything works fine. Many thanks already in advance for your time.